pkgs/development/tools/rust/cargo-llvm-cov/default.nix

```diff
@@ -1,36 +1,59 @@
 { stdenv
 , lib
-, fetchCrate
+, fetchurl
+, fetchFromGitHub
 , rustPlatform
+, rustc
 }:
 
-rustPlatform.buildRustPackage rec {
+let
   pname = "cargo-llvm-cov";
   version = "0.5.25";
 
-  src = fetchCrate {
-    inherit pname version;
-    sha256 = "sha256-4ctwvDLluJsLWJPInoFGqxmEzlEuBtEJb3/x+q/5pDA=";
+  owner = "taiki-e";
+  homepage = "https://github.com/${owner}/${pname}";
+
+  llvm = rustc.llvmPackages.llvm;
+
+  # Download `Cargo.lock` from crates.io so we don't clutter up Nixpkgs
+  cargoLock = fetchurl {
+    name = "Cargo.lock";
+    url = "https://crates.io/api/v1/crates/${pname}/${version}/download";
+    sha256 = "sha256-6VLAno4XhThzf8zdxWo32tfaHvOpHGOFwoDxMfm9ONs=";
+    downloadToTemp = true;
+    postFetch = ''
+      tar xzf $downloadedFile ${pname}-${version}/Cargo.lock
+      mv ${pname}-${version}/Cargo.lock $out
+    '';
+  };
+in
+
+rustPlatform.buildRustPackage {
+  inherit pname version;
+
+  # Use `fetchFromGitHub` instead of `fetchCrate` because the latter does not
+  # pull in fixtures needed for the test suite
+  src = fetchFromGitHub {
+    inherit owner;
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-+ddGNS2NcYNslkgd6DmJr0gtEvF49oOWTeDoNMJ8yIc=";
   };
-  cargoSha256 = "sha256-QghbQYfoCd+ppNz/g5NlCnrFYpsjesQlcgMCEKUgN2k=";
-
-  # skip tests which require llvm-tools-preview
-  checkFlags = [
-    "--skip bin_crate"
-    "--skip cargo_config"
-    "--skip clean_ws"
-    "--skip instantiations"
-    "--skip merge"
-    "--skip merge_failure_mode_all"
-    "--skip no_test"
-    "--skip open_report"
-    "--skip real1"
-    "--skip show_env"
-    "--skip virtual1"
-  ];
-
-  meta = rec {
-    homepage = "https://github.com/taiki-e/${pname}";
+
+  # Upstream doesn't include the lockfile so we need to add it back
+  postUnpack = ''
+    cp ${cargoLock} source/Cargo.lock
+  '';
+
+  cargoSha256 = "sha256-eYNb3FJkOxwyWAs8fEyQOEqpD3Fw5tCokdh5ghO8kro=";
+
+  # `cargo-llvm-cov` reads these environment variables to find these binaries,
+  # which are needed to run the tests
```

Comment:
I don't think it should go here, but can you add a similar note about why the tests would break in the future (referencing the conversations we've had here + the upstream issue I filed)?

Reply:
Done in a new commit, lmk whether what I wrote is adequate